src/jit/importer.cpp

```diff
@@ -6958,7 +6958,7 @@ bool  Compiler::impMethodInfo_hasRetBuffArg(CORINFO_METHOD_INFO * methInfo)
     //
   #ifdef FEATURE_HFA
     // On ARM HFAs are returned in registers.
-    if (!info.compIsVarArgs && IsHfa(methInfo->args.retTypeClass))
+    if (!info.compIsVarArgs && !opts.compUseSoftFP && IsHfa(methInfo->args.retTypeClass))
     {
```

Comment:
For now you can modify the above method to return false when compUseSoftFP is enabled,
But ultimately the ifdef FEATURE_HFA might need to be undef-ed when building the VM and JIT.
Currently the VM marks certain structs as HFA's and may treat them differently in Interop code.